services/addFilter.js

```diff
@@ -0,0 +1,20 @@
+export default function addFilter(array, filter) {
```

Comment:
```suggestion
export default function addFilter(array, filter = {}) {
```

by giving filter a default value you can drop all the ? after the filter because it is at least an empty object.